edgecontext/middlewares.go

```diff
@@ -46,3 +40,24 @@ func InjectHTTPEdgeContext(logger log.Wrapper) endpoint.Middleware {
 		}
 	}
 }
+
+// InitializeThriftEdgeContext sets an edge request context created from the
+// Thrift headers set on the context onto the context.
+func InitializeThriftEdgeContext(ctx context.Context, logger log.Wrapper) context.Context {
+	return InitializeEdgeContext(ctx, logger, FromThriftContext)
+}
+
+// InjectThriftEdgeContext returns a go-kit endpoint.Middleware that injects an
+// edge request context created from the Thrift headers set on the context into
+// the `next` endpoint.Endpoint.
+//
+// Note, this depends on the edge context headers already being set on the
+// context object.  You can configure the thrift server to do this by passing
+// thriftbp.HeadersToForward to the server's SetForwardHeaders method.
```

Comment:
`SetForwardHeaders` only affects how this server will forward headers to upstreams. For this to work that's not required. thrift's TSimpleServer implementation already injected it for you.